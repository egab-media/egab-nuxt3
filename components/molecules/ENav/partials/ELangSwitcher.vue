<template>
  <v-select
    v-model="localeSelection"
    :items="languages"
    dense
    background-color="transparent"
    flat
    class="mt-5"
    item-text="value.name"
    item-value="value.code"
    full-width
    title="select language"
    aria-label="select language"
    :menu-props="{ bottom: true, offsetY: true }"
  >
    <template #item="{ item }">
      <v-list-item dense :prepend-avatar="item.value.flag" :to="switchLocalePath(item.value.code)" @click="$i18n.setLocale(item.value.code)">
        <v-list-item-title :class="item.value.name.toLowerCase() === 'arabic' ? 'atom-font__arabic' : ''">
          {{ item.value.name === 'Arabic' ? 'العربية' : item.value.name }}
        </v-list-item-title>
      </v-list-item>
    </template>
    <template #selection="{ item }">
      <v-list-item dense class="pa-0" :prepend-avatar="item.value.flag">
      <v-list-item-title class="text-body-2 font-weight-bold">
        {{ item.value.name === 'Arabic' ? 'العربية' : item.value.name }}
      </v-list-item-title>
      </v-list-item>
    </template>
  </v-select>
</template>

<script lang="ts">
export default defineComponent({
  name: 'ELangSwitcher'
})
</script>

<script lang="ts" setup>
import {VSelect} from "vuetify/components/VSelect";
import { VListItem, VListItemTitle } from 'vuetify/components/VList'
import {useLanguageStore} from "~/store/language";

const { locale } = useI18n()

const {languages} = useLanguageStore()
const localeSelection = ref(locale.value)
</script>

<style scoped>

</style>
