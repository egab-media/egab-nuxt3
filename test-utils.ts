import { vi } from 'vitest'
import en from '@/locales/en-US.json'
import ar from '@/locales/ar-SA.json'
import { createTestingPinia } from '@pinia/testing'
import { createVuetify } from 'vuetify'
import * as vueTestUtils from '@vue/test-utils'
import { createApp } from 'vue'
import { createI18n } from 'vue-i18n'

export const addVuetify = (context: any) => {
    context.vuetifyInstance = createVuetify({});
    context.vueTestUtils.config.global.plugins = [context.vuetifyInstance];
    context.vueTestUtils.config.global.CSS = { supports: () => false };
}

// export const addVuex = (context: { vuex: any; vue: { use: (arg0: any) => void } }) => {
//     context.vuex = require('vuex')
//     context.vue.use(context.vuex)
// }

export const addPinia = (context: any) => {
    context.vueTestUtils.config.global.plugins.push(createTestingPinia())
}

export const addI18n = (context: any): any => {
    // eslint-disable-next-line new-cap
    context.i18nInstance = createI18n({
        // ...options,
        messages: { en, ar },
        fallbackLocale: 'en',
        legacy: true,
        globalInjection: true,
    })
    context.vueTestUtils.config.global.plugins.push(context.i18nInstance)
}

export const addFilter = (name: any, lambda: any): any => {
    return (context: { vue: any[] }) => context.vue.filter(name, lambda)
}

export const compositeConfiguration = (...configs: any[]): any => {
    return (context: any) => configs.forEach(config => config(context))
}

export const bootstrapVueContext = (configureContext: (arg0: {}) => any) => {
    const context: any = {}

    const teardownVueContext = () => {
        vi.unmock('vue')
        Object.keys(context).forEach(key => delete context[key])
        vi.resetModules()
    }

    context.vueTestUtils = vueTestUtils
    context.vue = createApp({})

    vi.doMock('vue', () => context.vue)

    configureContext && configureContext(context)

    return {
        teardownVueContext,
        ...context
    }
}

export const shallowMount = (component: any, context: any) => {
    return context.vueTestUtils.shallowMount(component, {
        localVue: context.localVue,
        vuetify: context.vuetifyInstance,
        i18n: context.i18nInstance,
        store: context.store,
        router: context.routerInstance,
        propsData: context.propsData,
        stubs: context.stubs,
        mocks: context.mocks,
        slots: context.slots,
        attachTo: context.attachTo
    })
}

export const mountWrapper = (component: any, context: any) => {
    return context.vueTestUtils.mount(component, {
        localVue: context.localVue,
        vuetify: context.vuetifyInstance,
        i18n: context.i18nInstance,
        store: context.store,
        router: context.routerInstance,
        propsData: context.propsData,
        stubs: context.stubs,
        mocks: context.mocks,
        slots: context.slots,
        attachTo: context.attachTo
    })
}
