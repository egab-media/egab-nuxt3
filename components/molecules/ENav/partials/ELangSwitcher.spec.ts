import {VueWrapper} from "@vue/test-utils";
import { beforeEach, afterEach, describe, it, expect } from 'vitest'
import ELangSwitcher from './ELangSwitcher.vue'
import {
    addVuetify,
    bootstrapVueContext,
    compositeConfiguration,
    addI18n,
    addPinia,
    mountWrapper
} from "~/test-utils";

let wrapper: VueWrapper<any>;
let vueContext: any;

describe('ELangSwitcher', () => {
    beforeEach(() => {
        vueContext = bootstrapVueContext(compositeConfiguration(addVuetify, addI18n, addPinia))
        wrapper = mountWrapper(ELangSwitcher, vueContext)
    })
    afterEach(() => {
        vueContext.teardownVueContext()
    })

    it('should mount', () => {
        expect(wrapper.vm).toBeTruthy()
        const selection = wrapper.find('.v-select__selection').find('.v-list-item-title')
        expect(selection.text()).toEqual('English')
    })
})
