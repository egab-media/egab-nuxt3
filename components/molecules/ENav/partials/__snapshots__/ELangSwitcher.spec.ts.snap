// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ELangSwitcher > should try methods 1`] = `
"<div data-v-2bec278d=\\"\\" class=\\"v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--dirty v-text-field v-select v-select--single v-select--selected v-select--selection-slot mt-5\\">
  <!---->
  <div class=\\"v-input__control\\">
    <div class=\\"v-field v-field--active v-field--appended v-field--center-affix v-field--dirty v-field--flat v-field--no-label v-field--variant-filled v-theme--light v-locale--is-ltr\\" role=\\"button\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-2\\">
      <div class=\\"v-field__overlay\\"></div>
      <div class=\\"v-field__loader\\">
        <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
          <!---->
          <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
          <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
            <div class=\\"v-progress-linear__indeterminate\\">
              <div class=\\"v-progress-linear__indeterminate long\\"></div>
              <div class=\\"v-progress-linear__indeterminate short\\"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <div class=\\"v-field__field\\" data-no-activator=\\"\\">
        <!----><label class=\\"v-label v-field-label\\" for=\\"input-0\\">
          <!---->
          <!---->
        </label>
        <!---->
        <div class=\\"v-field__input\\" data-no-activator=\\"\\">
          <!---->
          <!---->
          <div class=\\"v-select__selection\\">
            <div data-v-2bec278d=\\"\\" class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--variant-text pa-0\\" style=\\"max-height: 10px;\\" dense=\\"\\">
              <!----><span class=\\"v-list-item__underlay\\"></span>
              <div class=\\"v-list-item__prepend\\">
                <nuxt-img data-v-2bec278d=\\"\\" src=\\"https://flagcdn.com/w40/us.png\\" width=\\"40px\\" height=\\"25px\\"></nuxt-img>
                <div class=\\"v-list-item__spacer\\"></div>
              </div>
              <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
                <!---->
                <!---->
                <div data-v-2bec278d=\\"\\" class=\\"v-list-item-title text-body-2 font-weight-bold\\">English</div>
              </div>
              <!---->
            </div>
          </div><input size=\\"1\\" type=\\"text\\" id=\\"input-0\\" aria-describedby=\\"input-0-messages\\" inputmode=\\"none\\" aria-label=\\"Open\\" title=\\"Open\\" dense=\\"\\" cols=\\"12\\" background-color=\\"transparent\\" item-text=\\"value.name\\" full-width=\\"\\">
        </div>
        <!---->
      </div>
      <!---->
      <div class=\\"v-field__append-inner\\">
        <!----><i class=\\"mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-select__menu-icon\\" aria-hidden=\\"true\\"></i>
        <!---->
      </div>
      <div class=\\"v-field__outline\\">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <!---->
  <div class=\\"v-input__details\\">
    <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-0-messages\\">
      <!---->
    </transition-group-stub>
    <!---->
  </div>
</div>"
`;
